flux-syntax/src/surface.rs

```diff
@@ -192,6 +194,8 @@ pub enum TyKind<R = ()> {
     Constr(Expr, Box<Ty<R>>),
     Tuple(Vec<Ty<R>>),
     Array(Box<Ty<R>>, ArrayLen),
+    /// The first `R` parameter is for the `DefId` corresponding to the hir OpaqueTy
+    Opaque(R, Option<Vec<GenericArg<R>>>, Bounds<R>),
```

Comment:
Let's call this `ImplTrait` to be consistent with [`rustc_ast::TyKind`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_ast/ast/enum.TyKind.html#variant.ImplTrait)